api/cpp/include/slint_point.h

```diff
@@ -22,15 +22,27 @@ struct Point
 };
 
 namespace cbindgen_private {
-// The Size types are expanded to the Point2D<...> type from the euclid crate which
+// The Point types are expanded to the Point2D<...> type from the euclid crate which
 // is binary compatible with Point<T>
 template<typename T>
 using Point2D = Point<T>;
 }
 
 /// A position in logical pixel coordinates
-using LogicalPosition = Point<float>;
+struct LogicalPosition: public Point<float> {
+    explicit LogicalPosition(const Point<float> p): Point<float>(p) { };
+    LogicalPosition() : Point<float>{0., 0.} {};
+    LogicalPosition(const LogicalPosition&) = default;
+    LogicalPosition(LogicalPosition&&) = default;
+    ~LogicalPosition() = default;
```

Comment:
I believe you should leave that out.
(rule of 5 or rule of 0,  in this case, pick rule of 0)

Unless there is something i'm missing.  And in that case you should also add the default operator=

```suggestion
```

Reply:
I need one constructor for the conversion from Size<...>/Point<...>. That disables all the default constructors, so I have to add them back in.

So rule of 5 it is :-/

operator = added!

Reply:
> That disables all the default constructors

That disable only **the** default constructor.

The copy and move constructors are not disabled.